### ISSUE TITLE
LAMB implementation not coherent with Algorithm 2 in paper

### ISSUE THREAD

**Issue by RickZack** [NUS-HPC-AI-Lab/pytorch-lamb#1]
Dear NUS HPC AI Lab,
I would like to use your implementation of LAMB optimizer in my research project, but I noticed a difference between what is indicated in algorithm 2 of the original paper _Large Batch Optimization For Deep Learning: Training Bert In 76 Minutes_ and the actual implementation provided in this repository.
In particular, when the weight decay is zero, the code seems to implement the Adam algorithm, without the rescaling `w_norm/g_norm` that is peculiar to the LARS algorithm. Its implementation in this same repository seems coherent with what was prescribed in the original paper.

Am I wrong?

**Comment by Edenzzzz** [comment]
I believe it should be like this 
<img width="936" alt="image" src="https://github.com/NUS-HPC-AI-Lab/pytorch-lamb/assets/87317405/a9079e52-f1e5-41a8-8fe7-830178fd53b3">
